homeassistant/components/keymitt_ble/__init__.py

```diff
@@ -0,0 +1,121 @@
+"""Integration to integrate MicroBot with Home Assistant."""
+from __future__ import annotations
+
+import asyncio
+import contextlib
+import logging
+from typing import TYPE_CHECKING, Any
+import async_timeout
+
+from microbot import (
+    MicroBotApiClient,
+    parse_advertisement_data,
+)
+
+from homeassistant.components import bluetooth
+from homeassistant.components.bluetooth.passive_update_coordinator import (
+    PassiveBluetoothDataUpdateCoordinator,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, ServiceCall, callback
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.const import CONF_ACCESS_TOKEN, CONF_DEVICE
+
+from .const import DOMAIN
+
+if TYPE_CHECKING:
+    from bleak.backends.device import BLEDevice
+
+_LOGGER: logging.Logger = logging.getLogger(__package__)
+PLATFORMS: list[str] = [Platform.SWITCH]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up this integration using UI."""
+    hass.data.setdefault(DOMAIN, {})
+    token: str = entry.data[CONF_ACCESS_TOKEN]
+    ble_device: BLEDevice | None = entry.data[CONF_DEVICE]
```

Comment:
BLEDevice objects are transient. They can't be stored.  You need to exchange the address for a BLEDevice object

Reply:
```python3
    ble_device = bluetooth.async_ble_device_from_address(hass, address)
    if not ble_device:
        raise ConfigEntryNotReady(
            f"Could not find Switchbot {sensor_type} with address {address}"
        )

```

Reply:
Ah thanks